docs/architecture.md

```diff
@@ -0,0 +1,144 @@
+---
+title: Architecture
+weight: 20
+description: >
+  Architecture of Seccomp Agent.
+---
+
+
+## From Kubernetes to Seccomp Agent
+
+![From Kubernetes to Seccomp Agent](kubernetes-to-seccomp-agent.png "From Kubernetes to Seccomp Agent")
+
+Steps:
+- A user creates a pod with `kubectl apply -f pod.yaml`.
+  The file `pod.yaml` specifies the seccomp profile `foo.json` as follows:
+  ```
+  apiVersion: v1
+  kind: Pod
+  metadata:
+    name: mypod
+  spec:
+    securityContext:
+      seccompProfile:
+        type: Localhost
+        localhostProfile: foo.json
+  ```
+- Once the pod is scheduled on a node, a Kubelet on a node receives the pod
+  specification.
+- The Kubelet calls the method
+  [StartContainer](https://github.com/kubernetes/cri-api/blob/v0.22.0-alpha.2/pkg/apis/runtime/v1/api.proto#L65)
+  on containerd using the
+  [Container Runtime Interface (CRI)](https://github.com/kubernetes/cri-api).
+  The method includes the field `SeccompProfilePath` with the value
+  `/var/lib/kubelet/seccomp/foo.json`.
+- Containerd reads the file `/var/lib/kubelet/seccomp/foo.json`:
+  ```
+  {
+    "listenerPath": "/run/seccomp-agent.socket",
+    "defaultAction": "SCMP_ACT_ALLOW",
+    "syscalls": [
+      {
+        "names": ["mount"],
+        "action": "SCMP_ACT_NOTIFY"
+      }
+  }
+  ```
+- Containerd prepares config.json that includes the file foo.json from above:
+  ```
+  {
+    "linux": {
+      "seccomp": {
+        "listenerPath": "/run/seccomp-agent.socket",
+        "defaultAction": "SCMP_ACT_ALLOW",
+        "syscalls": [
+          {
+            "names": ["mount"],
+            "action": "SCMP_ACT_NOTIFY"
+          }
+  ```
+- Containerd starts runc with `config.json`.
+- Runc gets the seccomp file descriptor by calling the seccomp() system call
+  via libseccomp-golang.
+- Runc sends the seccomp file descriptor to the seccomp agent, along with
+  following
+  [Container Process State](https://github.com/opencontainers/runtime-spec/blob/1c3f411f041711bbeecf35ff7e93461ea6789220/config-linux.md#containerprocessstate):
+  ```
+  {
+    "ociVersion": "0.2.0",
+    "fds": ["seccompFd"],
+    "pid": 4422,
+    "metadata": "",
+    "state": {
+      "ociVersion": "0.2.0",
+      "id": "8eea9f6d...",
+      "status": "creating",
+      "pid": 4422,
+      "bundle": "rootfs",
+      "annotations": {}
+    }
+  }
+  ```
+- At this point, the Seccomp Agent has the seccomp file descriptor and the
+  Container Process State contains the following
+  [annotations](https://github.com/containerd/containerd/blob/v1.5.2/pkg/cri/annotations/annotations.go)
+  to identify the pod. Additional information such as the ServiceAccount could
+  be queried to the Kubernetes API.
```

Comment:
Why the service account? It seems like a weird example to give. What about just removin that part?

```suggestion
to identify the pod. Additional information can be queried to the Kubernetes API.
```

Reply:
I can remove it.

I used the ServiceAccount because we could imagine that someone would use the seccomp agent to implement some security service that grants or denies actions based on the ServiceAccount (Pod Security Policy were linked to ServiceAccount too).